### PR TITLE
Use the text from point to bol for prefix when caching

### PR DESCRIPTION
**Pull request by yyoncho**
- Fixes https://github.com/emacs-lsp/lsp-java/issues/95 and https://github.com/emacs-lsp/lsp-java/issues/46

### CODE REVIEW

**Comment by yyoncho** [comment]
@tigersoldier did you had a chance to take a look?

File: company-lsp.el [hunk 1]
```diff
@@ -102,6 +102,10 @@ COMPLETION is a cache-item created by `company-lsp--cache-item-new'.")
   (let ((provider (lsp--capability "completionProvider")))
     (and provider (seq-into (gethash "triggerCharacters" provider) 'list))))
 
+(defun company-lsp--prefix-line ()
+  "Return the string from point to bol."
```
Comment (tigersoldier) [inline company-lsp.el]: nit: from bol to point

File: company-lsp.el [hunk 2]
```diff
@@ -230,7 +234,6 @@ CANDIDATE is a string returned by `company-lsp--make-candidate'."
          (item (company-lsp--candidate-item resolved-candidate))
          (prefix (company-lsp--candidate-prefix candidate))
```
Comment (tigersoldier) [inline company-lsp.el]: Can you rename `prefix` to `prefix-line`?

Also rename `company-lsp--candidate-prefix` to `company-lsp--candidate-prefix-line` and the symbol it's using for text properties. `company-lsp--make-candidate` should be updated accordingly as well.

File: company-lsp.el [hunk 3]
```diff
@@ -477,9 +481,5 @@ See the documentation of `company-backends' for COMMAND and ARG."
   (when company-lsp-enable-snippet
     '(:textDocument (:completion (:completionItem (:snippetSupport t))))))
 
-(add-hook 'lsp-before-initialize-hook
```
Comment (tigersoldier) [inline company-lsp.el]: This change is not related to this PR. Please revert it and send a separate one.

File: company-lsp.el [hunk 4]
```diff
@@ -459,11 +463,11 @@ See the documentation of `company-backends' for COMMAND and ARG."
      ;; actions only apply to the pre-completion content. We backup the current
      ;; prefix and restore it after company completion is done, so the content
      ;; is restored and textEdit actions can be applied.
-     (or (company-lsp--cache-item-candidates (company-lsp--cache-get arg))
+     (or (company-lsp--cache-item-candidates (company-lsp--cache-get (company-lsp--prefix-line)))
```
Comment (tigersoldier) [inline company-lsp.el]: Update the parameter name to `prefix-line` in `company-lsp--cache-get`, `company-lsp--candidates-async` and `company-lsp--candidates-sync`

File: company-lsp.el (hunk 2 above)
Comment (yyoncho) [reply company-lsp.el]: Fixed.

File: company-lsp.el (hunk 3 above)
Comment (yyoncho) [reply company-lsp.el]: Done.

File: company-lsp.el (hunk 4 above)
Comment (yyoncho) [reply company-lsp.el]: Done, (hopefully, I have fixed everything).

**Comment by yyoncho** [comment]
Pushed new version. I have extended the commit description, please let me know whether it is sufficient.

**Comment by tigersoldier** [comment]
I just realized that this change breaks client-side filtering, which relies on prefix
